pyccel/parser/syntax/openmp.py

```diff
@@ -31,142 +35,547 @@ def __init__(self, **kwargs):
         """
         self.stmt = kwargs.pop('stmt')
 
-        super(OpenmpStmt, self).__init__(**kwargs)
+        super().__init__(**kwargs)
 
     @property
     def expr(self):
         if DEBUG:
             print("> OpenmpStmt: expr")
 
         stmt = self.stmt
-        if isinstance(stmt, OmpEndClause):
-            return stmt.expr
-        elif isinstance(stmt, OmpParallelConstruct):
-            return stmt.expr
-        elif isinstance(stmt, OmpLoopConstruct):
-            return stmt.expr
-        elif isinstance(stmt, OmpSingleConstruct):
+        if isinstance(stmt, (OmpEndClause, OmpParallelConstruct, OmpLoopConstruct,
+                             OmpSingleConstruct, OmpCriticalConstruct, OmpBarrierConstruct,
+                             OmpMasterConstruct, OmpMaskedConstruct, OmpTaskLoopConstruct,
+                             OmpSimdConstruct, OmpAtomicConstruct, OmpTaskWaitConstruct,
+                             OmpTaskyieldConstruct, OmpTaskConstruct, OmpFlushConstruct,
+                             OmpCancelConstruct, OmpTargetConstruct, OmpTeamsConstruct,
+                             OmpDistributeConstruct, OmpSectionConstruct, OmpSectionsConstruct)):
             return stmt.expr
         else:
             raise TypeError('Wrong stmt for OpenmpStmt')
 
+def check_get_clauses(name, valid_clauses, clauses):
+    txt = ''
+    for clause in clauses:
+        if isinstance(clause, valid_clauses):
+            txt = '{0} {1}'.format(txt, clause.expr)
+        else:
+            raise TypeError('Wrong clause', type(clause).__name__,'for', type(name).__name__)
```

Comment:
Errors do not work like `print` statements. This will not concatenate your strings